### PR TITLE
SDI-85: 🐛 Fix check my diary 2fa bypass

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/service/MfaClientService.kt [hunk 1]
```diff
@@ -12,13 +13,13 @@ open class MfaClientService(
...
 ) {
 
-  open fun clientNeedsMfa(request: AuthorizationRequest?): Boolean {
+  open fun clientNeedsMfa(request: AuthorizationRequest?, user: UserDetails?): Boolean {
     val client = clientDetailsService.loadClientByClientId(request?.clientId)
     val baseClientId = ClientService.baseClientId(client.clientId)
 
     // Special case for migration of check my diary users - ignore 2fa if they don't have the migrated role
     if (baseClientId == "my-diary" &&
-      request?.authorities?.none { it.authority == "ROLE_CMD_MIGRATED_MFA" } == true
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/service/MfaClientService.kt:21]: realised that the `request` doesn't actually hold the user authorities, so need to pass in the `user` instead

File: src/main/resources/db/dev/data/auth-postgresql/V900_0__clients.sql [hunk 2]
```diff
@@ -1,5 +1,5 @@
 INSERT INTO oauth_client_details (client_id, access_token_validity, additional_information, authorities, authorized_grant_types, autoapprove, client_secret, refresh_token_validity, resource_ids, scope, web_server_redirect_uri)
-VALUES ('another-delete-test-client', '1200', '{}', null, 'client_credentials', 'read', '$2a$10$lBwbziQlLfiCnn8Kj1PfMujEcLdsJYlYSNJvBRO638gCYTS9yN0xm', null, null, 'read', null),
+VALUES ('another-delete-test-client', 1200, '{}', null, 'client_credentials', 'read', '$2a$10$lBwbziQlLfiCnn8Kj1PfMujEcLdsJYlYSNJvBRO638gCYTS9yN0xm', null, null, 'read', null),
```
Comment (petergphillips) [inline src/main/resources/db/dev/data/auth-postgresql/V900_0__clients.sql:2]: fix to seed data

**Review by andymarke — APPROVED** [review]
LGTM
